ipatests/test_integration/test_sssd.py

```diff
@@ -148,3 +149,33 @@ def test_is_user_filtered(self, user):
             dp_req = ("Looking up [{0}] in data provider".format(
                 self.users[user]['name']))
             assert not dp_req.encode() in sssd_log
+
+    def test_external_group_paging(self):
+        """SSSD should fetch external groups without any limit.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4058
+        1: Add external groups more than default limit.
+        2: Run the command id aduser@ADDOMAIN.COM
+        3: sssd should retrieve all the external groups.
+        """
+        dn = DN(('cn', 'config'))
+        conn = self.master.ldap_connect()
+        entry = conn.get_entry(dn)  # pylint: disable=no-member
+        orig_limit = entry.single_value.get('nsslapd-sizelimit')
+        new_limit = int(orig_limit) + 2
```

Comment:
Nitpicking, but could you rename the variable "number_of_groups"? "new_limit" is misleading since the test doesn't change the sizelimit.